src/os/bluestore/Allocator.cc

```diff
@@ -125,6 +126,8 @@ Allocator *Allocator::create(CephContext* cct, string type,
     return new HybridAllocator(cct, size, block_size,
       cct->_conf.get_val<uint64_t>("bluestore_hybrid_alloc_mem_cap"),
       name);
+  } else if (type == "zoned") {
```

Comment:
need to guard this branch with `HAVE_LIBZBC`. otherwise the program will fail to link, because `ZonedAllocator` is not defined anywhere.

```
../../../lib/libos.a(Allocator.cc.o): In function `Allocator::create(ceph::common::CephContext*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, long, long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/home/jenkins-build/build/workspace/ceph-pull-requests/src/os/bluestore/Allocator.cc:130: undefined reference to `ZonedAllocator::ZonedAllocator(ceph::common::CephContext*, long, long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
collect2: error: ld returned 1 exit status
src/test/objectstore/CMakeFiles/unittest_alloc.dir/build.make:143: recipe for target 'bin/unittest_alloc' failed
make[3]: *** [bin/unittest_alloc] Error 1
```

Reply:
Done.